### ISSUE TITLE
Display Improve readability of containsExactly when the order of elements does not match

### ISSUE THREAD

**Issue by vlsi** [assertj/assertj-core#2629]
#### Feature summary

```java
assertThat(Arrays.asList(1, 2, 3, 4)).containsExactly(1, 3, 2, 4);
```

yields the following error message:

```
java.lang.AssertionError: 
Actual and expected have the same elements but not in the same order, at index 1 actual element was:
  2
whereas expected element was:
  3
```

Note that `1` and `4` are not displayed in the error message.

**I suggest** that AssertJ should print the full expected and actual set of elements in case the order does not match.

For instance, when the value does not match, then AssertJ does print all the elements:

```java
assertThat(Arrays.asList(1, 2, 3, 4)).containsExactly(1, 2, 4, 4);
```

```
java.lang.AssertionError: 
Expecting actual:
  [1, 2, 3, 4]
to contain exactly (and in same order):
  [1, 2, 4, 4]
but some elements were not found:
  [4]
and others were not expected:
  [3]
```

#### Example

The following methods seems to be affected:

```java
org.assertj.core.api.AbstractIterableAssert#containsExactly(ELEMENT... values)
org.assertj.core.api.AbstractIterableAssert#containsExactlyElementsOf(Iterable<? extends ELEMENT> iterable)
```

**Comment by joel-costigliola** [comment]
Fair enough @vlsi.

**Comment by Giovds** [comment]
I would like to give this one a go!

**Comment by joel-costigliola** [comment]
Thanks @Giovds, assigned to you :)

**Comment by joel-costigliola** [comment]
Fixed in 41179e555463021737468f71785ecca5aac152bb
